### PR TITLE
Response timeout as configurable CLI argument

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Add --response-timeout as new cli argument to allow tuning on shared
network connections.

### CODE REVIEW

File: ethminer/MinerAux.h [hunk 1]
```diff
@@ -315,7 +315,29 @@ class MinerCLI
 		}
 		else if ((arg == "--work-timeout") && i + 1 < argc)
 		{
-			m_worktimeout = atoi(argv[++i]);
+			try
+			{
+				m_worktimeout = atoi(argv[++i]);
```
Comment (chfast) [inline ethminer/MinerAux.h]: `atoi` does not throw any exception. only `stoi` does.

Comment (AndreaLanfranchi) [reply ethminer/MinerAux.h]: Right. Too quick cut and paste

Better have stoi and handle error instead of setting a value to zero for invalid input.

**Comment by AndreaLanfranchi** [comment]
> Not sure what this achieves.

Timeouts occur not when messages lost but when message delay too much.
2 seconds timeout makes a lot of sense if we intend it as "the pool is taking too much to repond to my submission request". In reality, from my empiric observations, serious pools generally respond immediately and we have time to responses which is very close to ping roundtrip time (eg. with avg PING rtt of 40~50 ms I always see response times of 45~55 ms).
So we have to handle two matters: network delay (which is in avg 80% of response time) and those randomly taken extra ms taken by pool to double check random samples of submitted shares.
So 2 seconds, imho, must not be taken as absolute value as we're not able to measure how much time the pool has taken to respond and how much was the time spent on trip of packets from pool to client.
This is the timeout we're implementing.

Another complete matter is timeout at socket level but this is out of our control for two reasons: a RVTMEO does not invalidate connection but instead instructs the client to send RESEND request to the server, and, less obvious, apparently Boost::ASIO does not care about RCVTIMEO

**Comment by jean-m-cyr** [comment]
My claim is that these response timeouts are cause by an Ethminer bug, loosing or dropping a packet, rather than any sort of pool or network latency. We shall see...

**Comment by AndreaLanfranchi** [comment]
As you previosly said (which I agree 100%)

> Who knows what Boost::asio does ?

Either we rewrite with simple async socket or we have to trust in Boost

**Comment by AndreaLanfranchi** [comment]
Someone please mind re-reviewing ?
Thanks

**Pull request merged**